File: src/perception.cpp
#include <ros/ros.h>


int main(int argc, char** argv){
    // Initialise ROS
    ros::init(argc, argv, "perception_node");
    ros::NodeHandle n;

    // Set up subscribers/publishers here
    


    // Main loop
    ROS_INFO("perception is running");
    ros::Rate rate(30);
    while (ros::ok()) {


      rate.sleep();
      ros::spinOnce();
    }

    return 0;
}

File: src/perception.py
#!/usr/bin/env python3

import rospy

# Set up callback here

# Initialize ROS 
rospy.init_node("perception_node")


# Set up subscribers here

# Main loop
rospy.loginfo("Perception is running")
rate = rospy.Rate(30)
while not rospy.is_shutdown():


    # Rate limiting
    rate.sleep()









